Backend/app/rider-platform/rider-app/Main/src/Domain/Action/Dashboard/Merchant.hs

```diff
@@ -335,6 +337,8 @@ postMerchantSpecialLocationGatesUpsert _merchantShortId _city specialLocationId
       latitude <- fromMaybeM (InvalidRequest "Latitude field cannot be empty for a new gate") $ reqT.latitude <|> (mbGate <&> (.point.lat))
       longitude <- fromMaybeM (InvalidRequest "Longitude field cannot be empty for a new gate") $ reqT.longitude <|> (mbGate <&> (.point.lon))
       address <- fromMaybeM (InvalidRequest "Address cannot be empty for a new gate") $ reqT.address <|> (mbGate >>= (.address))
+      mbMerchant <- QM.findByShortId merchantShortId
```

Comment:
you can use mid and mocid from specialLoction instead of making these two queries